src/ast/passes/attachpoint_passes.cpp

```diff
@@ -570,6 +576,23 @@ AttachPointParser::State AttachPointParser::lex_attachpoint(
   return State::OK;
 }
 
+bool AttachPointParser::should_skip_attachpoint(AttachPoint &ap)
+{
+  auto matches = bpftrace_.probe_matcher_->get_matches_for_ap(ap);
+  if (matches.empty()) {
+    const auto missing_probes = bpftrace_.config_->missing_probes;
+    if (missing_probes == ConfigMissingProbes::warn) {
+      ap.addWarning() << "Skipping non exist "
```

Comment:
We don't need this additional warning message. We already have two with the location (with this added logic to skip) without it.

Reply:
Thanks, I found that deleting this part causes a problem: when missing_probes == warn, similar to ignore, the following program will not print any warning.

```
config = {
  missing_probes = warn;
}

rawtracepoint:vmlinux:xhci_inc_d*,
rawtracepoint:vmlinux:not_exist_tp,
{
	@ring = args->ring;
}
```

if keep this message:

```
$ sudo ./raw_tp.bt 
./raw_tp.bt:8:1-35: WARNING: Skipping non exist rawtracepoint vmlinux:not_exist_tp
rawtracepoint:vmlinux:not_exist_tp,
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Attached 1 probe
^C
```

what do you think?

Reply:
Ah, I see. Ok, then let's update the language a bit:
```
ap.addWarning() << "No matches for " << probetypeName(probetype(ap.provider)) << " " << ap.target << ":" << ap.func << ". Skipping."
```